amino.go

```diff
@@ -2,21 +2,29 @@ package amino
 
 import (
 	"bytes"
-	"encoding/binary"
-	"encoding/json"
-	"errors"
 	"fmt"
 	"io"
 	"reflect"
 	"time"
+
+	"encoding/binary"
+	"encoding/json"
+
+	"github.com/pkg/errors"
 )
 
-//----------------------------------------
-// Global methods for global sealed codec.
-var gcdc *Codec
+var (
+	// Global methods for global sealed codec.
+	gcdc *Codec
+
+	// we use this time to init. a zero value (opposed to reflect.Zero which gives time.Time{} / 01-01-01 00:00:00)
+	zeroTime time.Time
 
-// we use this time to init. a zero value (opposed to reflect.Zero which gives time.Time{} / 01-01-01 00:00:00)
-var zeroTime time.Time
+	// NotPointerErr is thrown when you call a method that expects a pointer, e.g. Unmarshal
+	NotPointerErr = errors.New("expected a pointer")
+
+	UnknownBasicTypeErr = fmt.Errorf("got unknown type: ")
```

Comment:
Is this error used anywhere?

<img width="571" alt="Screen Shot 2019-05-22 at 12 41 02 PM" src="https://user-images.githubusercontent.com/1173161/58195996-e6269800-7c8e-11e9-991b-0d4e97d59bbb.png">

Reply:
Good catch! That's a relict from another approach (actually wrapping each basic type in a struct instead of just adding the necessary bytes). Removed.

Reply:
This is an excellent read on errors and touches the different error packages too: https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully